File: 151001/Novik/RESTLab1/REST/Mapper/NewsMapper.cs
using AutoMapper;
using REST.Models.DTOs.RequestIn;
using REST.Models.DTOs.Responses;
using REST.Models.Entity;

namespace REST.Mapper;

public class NewsMapper : Profile
{
    public NewsMapper()
    {
        CreateMap<News, NewsResponseTo>();
        CreateMap<NewsRequestTo, News>();
    }
    
}

File: 151001/Yankova/RV2/src/main/java/com/bsuir/nastassiayankova/mappers/NewsMapper.java
package com.bsuir.nastassiayankova.mappers;

import com.bsuir.nastassiayankova.beans.entity.News;
import com.bsuir.nastassiayankova.beans.request.NewsRequestTo;
import com.bsuir.nastassiayankova.beans.response.NewsResponseTo;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(componentModel = "spring", uses = {MessageListMapper.class, TagListMapper.class, UserMapper.class})
public interface NewsMapper {
    @Mapping(source = "userId", target = "user.id")
    News toNews(NewsRequestTo newsRequestTo);
    @Mapping(source = "user.id", target = "userId")
    NewsResponseTo toNewsResponseTo(News news);
}
